### ISSUE TITLE
Camping Tests automated integration tests don't catch missing upstream gem issues

### ISSUE THREAD

**Issue by karloscarweber** [camping/camping#172]
It's happened a few times where upstream gems have a missing dependency or an error that appear when running camping locally or in a deployed environment, but the error isn't apparent in our automated tests, or GitHub action tests. The current one I'm running into is about the new logger system. I think this is related to the disconnect between the gem file and the gem spec runtime dependencies.

Some relevant errors:

```ruby
<internal:/Users/kow/.rvm/rubies/ruby-3.3.0/lib/ruby/site_ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:37:in `require': cannot load such file -- dry/logger (LoadError)
        from <internal:/Users/kow/.rvm/rubies/ruby-3.3.0/lib/ruby/site_ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:37:in `require'
        from /Users/kow/.rvm/gems/ruby-3.3.0/gems/camping-3.2.0/lib/camping/gear/firewatch.rb:4:in `<top (required)>'
        from <internal:/Users/kow/.rvm/rubies/ruby-3.3.0/lib/ruby/site_ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:37:in `require'
        from <internal:/Users/kow/.rvm/rubies/ruby-3.3.0/lib/ruby/site_ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:37:in `require'
        from /Users/kow/.rvm/gems/ruby-3.3.0/gems/camping-3.2.0/lib/camping/loads.rb:16:in `<top (required)>'
        from <internal:/Users/kow/.rvm/rubies/ruby-3.3.0/lib/ruby/site_ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:37:in `require'
        from <internal:/Users/kow/.rvm/rubies/ruby-3.3.0/lib/ruby/site_ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:37:in `require'
        from /Users/kow/.rvm/gems/ruby-3.3.0/gems/camping-3.2.0/lib/camping.rb:1:in `<top (required)>'
        from <internal:/Users/kow/.rvm/rubies/ruby-3.3.0/lib/ruby/site_ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:37:in `require'
        from <internal:/Users/kow/.rvm/rubies/ruby-3.3.0/lib/ruby/site_ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:37:in `require'
        from /Users/kow/.rvm/gems/ruby-3.3.0/gems/camping-3.2.0/bin/camping:5:in `<top (required)>'
        from /Users/kow/.rvm/gems/ruby-3.3.0/bin/camping:25:in `load'
        from /Users/kow/.rvm/gems/ruby-3.3.0/bin/camping:25:in `<main>'
        from /Users/kow/.rvm/gems/ruby-3.3.0/bin/ruby_executable_hooks:22:in `eval'
        from /Users/kow/.rvm/gems/ruby-3.3.0/bin/ruby_executable_hooks:22:in `<main>'
```

To resolve this issue we need to recreate it in a test environment in Github Actions, Then write tests to ensure the inclusion of all dependencies in a mock application environment. Like we need to test more than just Camping, but USING CAMPING. Current tests are mostly like Unit tests, not integration tests.

### Checklist
- [ ] Write new tests sequence that uses bundler to install camping and test it's features.
- [ ] Rewrite logging subsystem to work as middleware as expected.

**Comment by karloscarweber** [comment]
Another example of the issue outlined above is the brand new logger. Unit tests all pass for the Logger, but fail in a production environment. To resolve this issue we'll need to make the logger work in production environments and by default.

The issue with the logger is that it's being used as middleware, which is fine. But it's currently configured to provide a ready object instead of Class for the Rack middleware. The key parts to resolve this are:

- `/lib/server.rb:113`, Where the `logging_middleware` is overwritten from `/lib/rackup/server.rb:267`. I think the solution is to grab the lambda used in `rackup`, and adapt it to use our `Firewatch` settings when instantiating the logger.

**Comment by karloscarweber** [comment]
This is resolved with Camping 3.2.4.
